pyuvdata/uvdata/tests/test_uvdata.py

```diff
@@ -9543,3 +9695,954 @@ def test_rephase_to_time():
+    assert object_id == 0
+
+    # Tick the obs_time up by a day, see if the software will fetch additional
+    # coordinates and expand the existing ephem
+    obs_time += 1
+    object_dict = sma_mir._phase_object_dict_helper(
+        0, 0, None, None, None, None, 0, 0, 0, 0, "Mars", True, None, obs_time,
+    )
+
+    # Previously, everything else will have had a single point, but the new ephem (which
+    # covers 36 hours at 3 hour intervals) should have a lucky total of 13 points.
+    keycheck = ["object_lon", "object_lat", "object_vrad", "object_dist", "coord_times"]
+    for key in keycheck:
+        assert len(object_dict[key]) == 13
+
+
+@pytest.mark.parametrize("future_shapes", [True, False])
+def test_fix_phase(hera_uvh5, sma_mir, future_shapes):
+    """
+    Test the phase fixing method
+    """
+    if future_shapes:
+        hera_uvh5.use_future_array_shapes()
+        sma_mir.use_future_array_shapes()
+
+    # Check the one error condition that fix_phase raises
+    with pytest.raises(ValueError, match="Cannot run fix_phase on a multi-obj"):
+        sma_mir.fix_phase(use_ant_pos=False)
+
+    # Make some copies of the data
+    uv_in = hera_uvh5.copy()
+    uv_in_bad = hera_uvh5.copy()
+
+    # These values could be anything -- we're just picking something that we know should
+    # be visible from the telescope at the time of obs (ignoring horizon limits).
+    phase_ra = uv_in.lst_array[-1]
+    phase_dec = uv_in.telescope_location_lat_lon_alt[0] * 0.333
+
+    # Do the improved phasing on the dat set.
+    uv_in.phase(phase_ra, phase_dec)
+
+    # First test the case where we are using the old phase method with the uvws
+    # calculated from the antenna positions. Using fix phase here should be "perfect",
+    # since the uvws are completely recalculated from scratch.
+    uv_in_bad.phase(phase_ra, phase_dec, use_old_proj=True, use_ant_pos=True)
+    uv_in_bad.fix_phase(use_ant_pos=True)
+    assert uv_in == uv_in_bad
+
+    uv_in_bad = hera_uvh5.copy()
+    uv_in_bad.phase(phase_ra, phase_dec, use_old_proj=True, use_ant_pos=False)
+    uv_in_bad.fix_phase(use_ant_pos=False)
+
+    # We have to handle this case a little carefully, because since the old
+    # unphase_to_drift was _mostly_ accurate, although it does seem to intoduce errors
+    # on the order of a part in 1e5, which translates to about a tenth of a degree phase
+    # error in the test data set used here. Check that first, make sure it's good
+    assert np.allclose(uv_in.data_array, uv_in_bad.data_array, rtol=3e-4)
+
+    # Once we know the data are okay, copy over data array and check for equality btw
+    # the other attributes of the two objects.
+    uv_in_bad.data_array = uv_in.data_array
+    assert uv_in == uv_in_bad
+
+
+@pytest.mark.parametrize("future_shapes", [True, False])
+def test_multi_phase_multi_file(hera_uvh5, future_shapes):
+    """
+    Test what happends when we attempt to add together two multi-pha
```

Comment:
cliff hanger docstring!

Reply:
Dun, dun, duuuuuuuuuuuuun! Fixed in 8307632fc961dd0f19d18f6d7681eca7df175a18.